share/gpodder/extensions/rockbox_coverart.py

```
@@ -37,10 +52,194 @@ def on_episode_synced(self, device, episode):
+                                            self.config.convert_allow_upscale_art is False:
+                                        copyflag = True
+                                    elif img.format.upper() != device_match_filetype:
+                                        copyflag = True
+                                    try:
+                                        if img.info['progressive'] == 1:
+                                            copyflag = True
+                                    except:
+                                        pass  # expected result if baseline jpeg or png
+                            except OSError:
+                                logger.info("%s check image error!", device_art)
+                        # file does not exist, we will create it
+                        else:
+                            copyflag = True
+
+                        if copyflag is True:
+                            logger.info("%s %s" % (device_art, "copying"))
+                            try:
+                                # should we file lock the source file?
+                                with Image.open(episode_art) as img:
+                                    if img.height > int(self.config.convert_size)\
+                                            or self.config.convert_allow_upscale_art is True:
+                                        out = img.resize((int(self.config.convert_size), int(self.config.convert_size)))
+                                    else:
+                                        out = img.copy()
+                                    out.save(device_art)
+                            except OSError:
+                                logger.info("%s image error!", episode_art)
+                        else:
+                            logger.info("%s %s" % (device_art, "already exists"))
+
+                        device_lockfile.release()
```

I'm not super comfortable with `device_lockfile` not used [as a context manager or with a try catch](https://py-filelock.readthedocs.io/en/latest/api.html#filelock.BaseAsyncFileLock.acquire).
Let's look at the code once more after other comments are addressed.

---

it's the code after the `acquire()` that could throw and exception causing the lock not to be released.
I see that every branch has its own `try: except` so it's ok.

---

I'll put a comment in there saying to be careful when modifying that to catch all exceptions within there so that we always release the lock, or something to that effect.